### ISSUE TITLE
[Product page] Overly verbose product variant announcement

### ISSUE THREAD

**Issue by svinkle** [Shopify/dawn#125]
## Issue summary

When navigating through the Product variant options, each variant is announced in addition to the currently selected variant.

For example, using <kbd>Tab</kbd> with a screen reader, a size variant of "35" is announced along side all other variant sizes:
- 🗣 _"35, selected, radio button, 1 of 7, Size 36 37 38 39 40 41"_

In addition, when using the screen reader virtual cursor, the `fieldset` group is described with all variant labels included. This would lead to a confusing or frustrating user experience for screen reader users.

<details>
  <summary>Screenshots</summary>
<img width="737" alt="Screenshot of Dawn product page. Variant size selection is highlighted." src="https://user-images.githubusercontent.com/1392632/124634981-ed03cb00-de54-11eb-9bf7-85347a63bfb4.png">

</details>

#### HTML

```html
<fieldset class="js product-form__input">
  <legend class="form__label">
    Size
    <p>
      <input
        type="radio"
        id="template--14172034990102__main-Size <p>-0"
        name="Size <p>"
        value="35"
        form="product-form-template--14172034990102__main"
        checked=""
      />
      <label for="template--14172034990102__main-Size <p>-0">35</label>
      <!-- … -->
    </p>
  </legend>
</fieldset>
```

## Steps to reproduce

1. Go to a Dawn Product page with a variant options with any supported browser.
2. With an active screen reader, <kbd>Tab</kbd> to the variant options.

### Expected

- Variant options to be announced with their own label in addition to the `legend` text only.

### Actual

- Varian options are announced with all other variant labels.

## Recommendation

1. Adjust the markup to move the variant option group as a sibling to the `legend` element.
2. Remove the wrapping `p` element as this is not required.

With this change, the previous example would now be:
- 🗣 _"35, selected, radio button, 1 of 7, Size"_

#### HTML

```html
<fieldset class="js product-form__input">
  <legend class="form__label">Size</legend>
  <input
    type="radio"
    id="template--14172034990102__main-Size <p>-0"
    name="Size <p>"
    value="35"
    form="product-form-template--14172034990102__main"
    checked=""
  />
  <label for="template--14172034990102__main-Size <p>-0">35</label>
  <!-- … -->
</fieldset>
```

<!-- #### Diff

```diff

``` -->

## Specifications

- **Component:** Product
- **WCAG Principal:** Perceivable
- **WCAG SC:** [1.3.1 Info and Relationships](https://www.w3.org/WAI/WCAG21/Understanding/info-and-relationships.html)
- **Severity:** Medium
- **Effort:** Low

**Comment by PaulNewton** [comment]
@svinkle  For accessibility of long lists* - The approach worked out here may impact the reference docs on handling readout of other "list" type features in other templates that can be very long such as the list of cart items, customer order history, collection lists, collection filters,etc.

How could we present reading all the remaining options at once instead of removing it entirely? 
Beyond the first selected radio choice wouldn't this then end up being a very long set of options* users have to tab through to get a full read out ,_since they would not longer be all read at once by default_ , swapping one annoyance for another annoyance? 
 No matter which way this is done the verbosity shows up somewhere does it not. Manageable for a couple of options but seems like massive chore and frustration for users with dozens or more of options in both situations. 

_note: shopify has a current variant limit of 100 so there is at least that ceiling on the amount of tabbing in a product option_

For inputs in screen readers is there a convenience like is done with links on page to choose to be able to read all at once all the options in a big set of form inputs(radios,checkbox,multi-groups)?

If not ,
 what's the common behavior of screen readers for the **next** tab beyond the first radio option in a long list of form inputs?
 Does that then read only the next option that the user is tabbing too? 
 Or does **that** tab into the second item then    automatically, _or via prompt_, cause a readout of all remaining options so a user doesn't have to endlessly tab to discovery everything sequentially?

If no such convenience exists then with only the proposed change the remaining options wouldn't exist at all in the readout except one at a time; unlike **before** where the set of all choices would be appended to the **initial** readout giving the user an upfront idea of just how big a list they are about to interact with having to either tap tab or long press hold tab 🚀 🚀  to get to the end of a list they don't know the end of.

 So what behavior do screens reader users deserve for the **remaining**  entries in long lists?
 Do screen reader users expect|require , as a standard, having to tab through each item individually even up to 100 times for variants, or more! for other sets of things in themes?
 Or do they deserve the next interaction to read the remaining options?
 Or for that method of reading all options to exist somewhere for them ?skiplist? to be able to choose to cursor into and avoid tabitis hell in the main list of long options*.
🤔 Can this be handled by screen reader modes like is there a way to read all the options in browse mode but only the selected option in focus mode to handle this?
_Could me moot if constrained by a big expectation that tabbing past a radio input is supposed to **always** go into either the next radio or item in fieldset or leaves the fieldset meaning any "read all" would have to before the first radio or **after** the fieldset inputs which would be pointless if they just tabbed through the whole list unless they are tabbing backwards 🤷‍♂️ ._ 

Hypothetical readout , structure undecided
"🗣 "35, selected, radio button, 1 of 7, Size" 
 <kbd>tab</kbd> \> 🗣 "List of all Sizes, text , <?>, 7 total, Size 35 36 37 38 39 40 41"
 <kbd>tab</kbd> \> "🗣 "36, selected, radio button, 2 of 7, Size" 

_Final wording would probably "All Size Options" since inline S-pluralization of variant options isn't guaranteed success i.e. "List of all Finishes**s**", so just use the raw option name._



*If any element or interaction comes of this should we define what a long list is for screen readers?

**Comment by svinkle** [comment]
@PaulNewton Proper markup via `name` grouping attribute on each `input` would provide the ability for users to <kbd>Tab</kbd> to the first item in the list, then <kbd>Tab</kbd> away from the list if not interested to discover more. When the first item is in focus, users can <kbd>Arrow</kbd> into the list to discover more options. Screen readers would announce each item once at a time. This would be the expected interaction.

**Comment by svinkle** [comment]
Fixed at some point.
